tests/unit/Internal/Primitives/CallableTypeTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Facile\PhpCodec\Internal\Primitives;
+
+use Eris\TestTrait;
+use Facile\PhpCodec\Codecs;
+use Tests\Facile\PhpCodec\BaseTestCase;
+use Tests\Facile\PhpCodec\GeneratorUtils;
+
+class CallableTypeTest extends BaseTestCase
+{
+    use TestTrait;
+
+    public function testLaws(): void
+    {
+        $this
+            ->forAll(
+                GeneratorUtils::scalar(),
```

Comment:
Technically correct, but scalar are never callable. Here you should have a generator that provides both callable and non-callable. Something like

```php
Eris\Generator\oneOf(
  GeneratorUtils::scalar(),
  Eris\Generator\map(function($x): callable { return function () use ($x) { return $x; } }, GeneratorUtils::scalar())
)
```

Take a look at https://eris.readthedocs.io/en/latest/generators/composite.html#

Reply:
>  but scalar are never callable

I realized now that it's not true. However, it's very unlikely that a pseudorandom generator returns a string that is also the name of a function.